packages/server/src/server.js

```diff
@@ -369,6 +378,7 @@ export default class Server {
       await this.renderer.close()
     }
 
+    this.app.stack.forEach(this.unloadMiddleware)
```

Comment:
Small note: This might break metal as it doesn't has `stack` prop but anyway it will be broken for HMR functionality.

Reply:
What about `this.app.stack && this.app.stack.forEach(this.unloadMiddleware)` ?

Reply:
It would be probably safer :) But also current status is fine because a standard express/connect instance should expose stack